datafusion/core/src/physical_optimizer/optimizer.rs

```diff
@@ -103,6 +103,10 @@ impl PhysicalOptimizer {
             // replacing operators with fetching variants, or adding limits
             // past operators that support limit pushdown.
             Arc::new(LimitPushdown::new()),
+            // The EliminateCommonPhysicalSubExprs rule extracts common physical
+            // subexpression trees into a `ProjectionExec` node under the actual node to
+            // calculate the common values only once.
+            Arc::new(EliminateCommonPhysicalSubexprs::new()),
```

Comment:
Do you have any examples of where this rule does eliminations for a plan that has been optimized by the corresponding logical rule?

Because if not it not clear to me why it should be part of the default/recommended list of rules.

Reply:
I think the usecase is for systems (like Comet) that don't use the LogicalPlanner.

That being said, I think the question of "should it be run by default" is a good one -- and I agree with your assertion if we can't find an example where this helps a plan we should probably not enable it by default.

FYI @andygrove

Reply:
Yes, this rule is useful only for Comet and similar usecases. Let me remove it from the default optimizer: https://github.com/apache/datafusion/pull/13046/commits/de98385bcec22e93f787bf51ce9cd8c304fd143d